src/resolver.js

```diff
@@ -0,0 +1,43 @@
+const github = require('@actions/github');
+
+const ISSUE_TYPES = ['issue_comment'];
+
+const getPrUrl = (payload) => {
+  try {
+    if (ISSUE_TYPES.includes(payload.event_name)) {
+      return github.context.payload.issue.pull_request.url;
+    }
+
+    return github.context.payload.pull_request.url;
+  } catch {
+    return null;
+  }
+};
+
+const getFallbackRefs = () => {
+  return {
+    status: null,
+    headRef: process.env.GITHUB_HEAD_REF,
+    baseRef: process.env.GITHUB_BASE_REF,
+  };
+};
+
+const resolveRefs = async (token) => {
+  const prUrl = getPrUrl(github.context.payload);
+
+  if (!prUrl) {
+    return getFallbackRefs();
+  }
```

Comment:
I'd remove `getFallbackRefs` because we never want to call it for real. I'd rather make `getPrUrl` iterate over supported `ISSUE_TYPES` and `throw` if there isn't a match. For each supported type we must know the corresponding key where refs are.